### ISSUE TITLE
user registration error

### ISSUE THREAD

**Issue by neerajdixit** [colinskow/superlogin#10]
Hi,

If I try to register a user on local secured couchdb instance with "userDBs" option. It fails with

[SyntaxError: Unexpected token u]
[SyntaxError: Unexpected token u]
SyntaxError: Unexpected token u
    at Object.parse (native)
    at /home/dixitne/work/nodetest/node_modules/superlogin/lib/dbauth/index.js:245:39
------------------------------------------------------------------------------------------------------------------------------
Here is my sample config

var config = {
  dbServer: {
    cloudant: false,
    protocol: 'http://',
    host: 'localhost:5984',
    user: 'test1',
    password: 'test1234',
    userDB: 'sl-users',
    couchAuthDB: '_users'
  },
  userDBs: {
    defaultDBs: {
      private: ['test']
    }
  }
}

I tried to debug and the problem is with the way it tries to send request to create the user on localDB which in turn is because of the way superagent.js client parses the url.

I was able to get this working for local env by changing the "getDBURL()" in util.js to not add username and password to the url instead use

.auth('couchdb_admin_username', 'couchdb_admin_password') 

in "createDB()"

I hope this helps in fixing the problem

Thanks,
Neeraj

**Comment by colinskow** [comment]
Hi @neerajdixit , thank you for reporting this. It could be that the username and password need to be URL encoded first if they are using any non-url characters. Go into `/node_modules/superlogin/lib/util.js`, line 50 and change it to:

```js
url = db.protocol + encodeURIComponent(db.user) + ':' + encodeURIComponent(db.password) + '@' + db.host;
```

Let me know if this fixes the problem. Otherwise I will do it your way, but that involves more code.

**Comment by colinskow** [comment]
But before you do that, go to `/node_modules/superlogin/lib/dbauth/index.js`, line 245 and change it to:
```js
return BPromise.reject(err.text);
```

(remove JSON.parse) so we can know what the exact error is.

**Comment by neerajdixit** [comment]
what you suggested works perfectly and yeah the JSON.parse was a problem to get exact error that I forgot to mention.

thanks..

**Comment by colinskow** [comment]
OK, I will create a release with the fix.

**Comment by colinskow** [comment]
The error was an illegal character in the URL?

**Comment by colinskow** [comment]
Should be fixed in SuperLogin 0.4.2. Let me know if you have any further problems.

**Comment by neerajdixit** [comment]
thanks. I did have some special characters in the password.
